ext/liquid_c/context.h

```diff
@@ -0,0 +1,48 @@
+#if !defined(LIQUID_CONTEXT_H)
+#define LIQUID_CONTEXT_H
+
+void init_liquid_context();
+VALUE context_evaluate(VALUE self, VALUE expression);
+VALUE context_find_variable(VALUE self, VALUE key, VALUE raise_on_not_found);
+void context_maybe_raise_undefined_variable(VALUE self, VALUE key);
+
+extern ID id_aset, id_call, id_to_liquid, id_set_context;
+
+#ifndef RB_SPECIAL_CONST_P
+// RB_SPECIAL_CONST_P added in Ruby 2.3
+#define RB_SPECIAL_CONST_P SPECIAL_CONST_P
+#endif
+
+inline static VALUE value_to_liquid_and_set_context(VALUE value, VALUE context_to_set)
+{
+    if (RB_SPECIAL_CONST_P(value))
```

Comment:
can you explain this?

Reply:
Yeah this function could use some comments. Every type that this matches has `def to_liquid; self end` so this avoid dispatch into Ruby for a noop. Though I forgot about https://github.com/Shopify/liquid/commit/bf3e759da3fc64de33233211d6246c6613ae6681, need to add a more complete test to make sure it works in exactly the same way.